gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+    FAILED_PRECONDITION(400, "global", "failedPrecondition", "FAILED_PRECONDITION"),
+    // change invalid argument error code to 412 when #440 is fixed
+    INVALID_ARGUMENT(400, "global", "badRequest", "INVALID_ARGUMENT"),
+    BAD_REQUEST(400, "global", "badRequest", "BAD_REQUEST"),
+    INTERNAL_ERROR(500, "global", "internalError", "INTERNAL_ERROR");
+
+    private final int code;
+    private final String domain;
+    private final String reason;
+    private final String status;
+
+    Error(int code, String domain, String reason, String status) {
+      this.code = code;
+      this.domain = domain;
+      this.reason = reason;
+      this.status = status;
+    }
+
+    Response response(String message) {
+      try {
+        return new Response(code, toJson(message));
+      } catch (IOException e) {
+        return Error.INTERNAL_ERROR.response("Error when generating JSON error response");
+      }
+    }
+
+    private String toJson(String message) throws IOException {
+      Map<String, Object> args = new HashMap<>();
+      Map<String, Object> errors = new HashMap<>();
+      errors.put("domain", domain);
+      errors.put("message", message);
+      errors.put("reason", reason);
+      args.put("errors", ImmutableList.of(errors));
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      return jsonFactory.toString(ImmutableMap.of("error", args));
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      // see https://cloud.google.com/resource-manager/reference/rest/
+      String path = exchange.getRequestURI().getPath();
+      String requestMethod = exchange.getRequestMethod();
+      Response response = null;
+      if (requestMethod.equals("POST") && path.startsWith("/" + VERSION + "/projects")) {
+        if (path.contains("undelete")) {
+          try {
+            response = undelete(projectIdFromURI(path));
+          } catch (IOException e) {
+            response = Error.BAD_REQUEST.response(e.getMessage());
+          }
+        } else {
+          String requestBody =
+              decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+          response = create(jsonFactory.fromString(requestBody, Project.class));
+        }
+      } else if (requestMethod.equals("DELETE")) {
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET")) {
+        if (path.startsWith("/" + VERSION + "/projects/")) {
+          response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+        } else {
+          response = list(parseListOptions(exchange.getRequestURI().getQuery()));
+        }
+      } else if (requestMethod.equals("PUT")) {
+        String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
```

Comment:
Nit, you should probably check that request path has the same projectId as in the body.